runtime/particle.js

```diff
@@ -177,6 +177,19 @@ export class Particle {
     }
     return false;
   }
+  _matchTypes(connection, handle) {
```

Comment:
It seems worth adding unit tests for this logic even though it's a private method. Is it a private method, in fact? It's used in `Multiplexer.js`. Maybe it shouldn't have the '_' prefix?

Reply:
refactored and added test. thanks!